crates/ethereum/payload/src/lib.rs

```diff
@@ -253,7 +251,7 @@ where
         // the EIP-4844 can still fit in the block
         if let Some(blob_tx) = tx.as_eip4844() {
             let tx_blob_gas = blob_tx.blob_gas();
-            if sum_blob_gas_used + tx_blob_gas > MAX_DATA_GAS_PER_BLOCK {
+            if sum_blob_gas_used + tx_blob_gas > BlobParams::cancun().max_blob_count {
```

Comment:
we need to change what we track here, we're now comparing gas used with blob count

gas used is `blob count * gas per blob`

so what we should track instead of `sum_blob_gas_used + tx_blob_gas` is simply the blob count